CoreArrangementPlugins/src/au/gov/asd/tac/constellation/plugins/arrangements/scatter3d/Scatter3dArranger.java

```diff
@@ -0,0 +1,250 @@
+ * limitations under the License.
+ */
+package au.gov.asd.tac.constellation.plugins.arrangements.scatter3d;
+
+import au.gov.asd.tac.constellation.graph.Graph;
+import au.gov.asd.tac.constellation.graph.GraphElementType;
+import au.gov.asd.tac.constellation.graph.GraphWriteMethods;
+import au.gov.asd.tac.constellation.graph.attribute.FloatAttributeDescription;
+import au.gov.asd.tac.constellation.graph.schema.analytic.attribute.objects.RawData;
+import au.gov.asd.tac.constellation.graph.schema.visual.concept.VisualConcept;
+import au.gov.asd.tac.constellation.plugins.arrangements.Arranger;
+import au.gov.asd.tac.constellation.plugins.arrangements.GraphUtilities;
+import au.gov.asd.tac.constellation.utilities.color.ConstellationColor;
+import java.time.ZonedDateTime;
+import java.util.BitSet;
+
+/**
+ * Arrange the graph in a scatter3d.
+ *
+ * @author CrucisGamma
+ */
+public class Scatter3dArranger implements Arranger {
+
+    private final Scatter3dChoiceParameters params;
+
+    /**
+     * A scatter3d arrangement with default parameters.
+     */
+    public Scatter3dArranger() {
+        this(Scatter3dChoiceParameters.getDefaultParameters());
+    }
+
+    /**
+     * Construct new ArrangeInScatter3d instance.
+     *
+     * @param params Parameters for the arrangement.
+     */
+    public Scatter3dArranger(final Scatter3dChoiceParameters params) {
+        this.params = params;
+    }
+
+    @Override
+    public void setMaintainMean(final boolean b) {
+    }
+
+    @Override
+    public void arrange(final GraphWriteMethods wg) throws InterruptedException {
+
+        // Get/set the x,y,z attributes.
+        if (wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.X.getName()) == Graph.NOT_FOUND) {
+            wg.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "x", "x", null, null);
+        }
+        if (wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.Y.getName()) == Graph.NOT_FOUND) {
+            wg.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "y", "y", null, null);
+        }
+        if (wg.getAttribute(GraphElementType.VERTEX, VisualConcept.VertexAttribute.Z.getName()) == Graph.NOT_FOUND) {
+            wg.addAttribute(GraphElementType.VERTEX, FloatAttributeDescription.ATTRIBUTE_NAME, "z", "z", null, null);
+        }
+
+        final String xDimension;
+        final String yDimension;
+        final String zDimension;
+        final boolean xLogarithmic;
+        final boolean yLogarithmic;
+        final boolean zLogarithmic;
+        if (params != null) {
+            xDimension = params.getXDimension();
+            yDimension = params.getYDimension();
+            zDimension = params.getZDimension();
+            xLogarithmic = params.getLogarithmicX();
+            yLogarithmic = params.getLogarithmicY();
+            zLogarithmic = params.getLogarithmicZ();
+        } else {
+            return;
```

Comment:
Did you mean to do nothing and return if no parameters were provided?

Reply:
Yes.  If the attributes aren't chosen, this should leave the graph as it stands at the present time.